File: 1-TwoSum/1-TwoSum.py
# Solution three - O(n) time and o(n) space - one O(n) itteration
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        vals = {}

        # For each number, it checks if the complment is in the hashmap already - if not it adds it
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in vals:
                return [i, vals[complement]]

            vals[nums[i]] = i

File: 1-TwoSum/1-TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> vals = new HashMap<>();
        ArrayList<Integer> ans = new ArrayList<>();
        
        for (int i = 0; i < nums.length; i ++){
            int complement = target - nums[i];

            System.out.println(complement);

            if(vals.containsKey(complement)){
                return new int[] {i, vals.get(complement)};
            }
            vals.put(nums[i], i);
        }
        return new int[] {};

    }
}